Index: ubuntu/Makefile
===================================================================
--- ubuntu.orig/Makefile	2009-09-02 16:42:09.067021527 -0500
+++ ubuntu/Makefile	2009-09-03 18:49:52.563795376 -0500
@@ -41,17 +41,17 @@
 		(cd $$subdir && $(MAKE) $@) || exit $$? ; done
 
 install: deploy
-	@$(INSTALL) -d $(prefix)
-	@$(INSTALL) -d $(etcdir)/eucalyptus/cloud.d
-	@$(INSTALL) -m 0644 VERSION $(etcdir)/eucalyptus/eucalyptus-version
-	@$(INSTALL) -d $(etcdir)/init.d
-	@$(INSTALL) -d $(vardir)/run/eucalyptus/net
-	@$(INSTALL) -d $(vardir)/lib/eucalyptus/keys
-	@$(INSTALL) -d $(vardir)/lib/eucalyptus/CC
-	@$(INSTALL) -d $(vardir)/log/eucalyptus
-	@$(INSTALL) -d $(datarootdir)/eucalyptus
-	@$(INSTALL) -d $(usrdir)/sbin
-	@$(INSTALL) -d $(usrdir)/lib/eucalyptus
+	@$(INSTALL) -d $(DESTDIR)$(prefix)
+	@$(INSTALL) -d $(DESTDIR)$(etcdir)/eucalyptus/cloud.d
+	@$(INSTALL) -m 0644 VERSION $(DESTDIR)$(etcdir)/eucalyptus/eucalyptus-version
+	@$(INSTALL) -d $(DESTDIR)$(etcdir)/init.d
+	@$(INSTALL) -d $(DESTDIR)$(vardir)/run/eucalyptus/net
+	@$(INSTALL) -d $(DESTDIR)$(vardir)/lib/eucalyptus/keys
+	@$(INSTALL) -d $(DESTDIR)$(vardir)/lib/eucalyptus/CC
+	@$(INSTALL) -d $(DESTDIR)$(vardir)/log/eucalyptus
+	@$(INSTALL) -d $(DESTDIR)$(datarootdir)/eucalyptus
+	@$(INSTALL) -d $(DESTDIR)$(usrdir)/sbin
+	@$(INSTALL) -d $(DESTDIR)$(usrdir)/lib/eucalyptus
 	@for subdir in $(SUBDIRS); do \
 		(cd $$subdir && $(MAKE) $@) || exit $$? ; done
 
Index: ubuntu/clc/modules/bootstrap/Makefile
===================================================================
--- ubuntu.orig/clc/modules/bootstrap/Makefile	2009-09-02 16:42:09.083018115 -0500
+++ ubuntu/clc/modules/bootstrap/Makefile	2009-09-03 18:49:52.728777511 -0500
@@ -45,8 +45,8 @@
 	gengetopt --input=arguments.ggo --file-name=eucalyptus-opts --func-name=arguments --arg-struct-name=eucalyptus_opts
 
 install:
-	$(INSTALL) -d $(sbindir)
-	$(INSTALL) -m 0755 eucalyptus-cloud $(sbindir)/eucalyptus-cloud
+	$(INSTALL) -d $(DESTDIR)$(sbindir)
+	$(INSTALL) -m 0755 eucalyptus-cloud $(DESTDIR)$(sbindir)/eucalyptus-cloud
 
 clean:
 	rm -f $(OBJS) eucalyptus-cloud
Index: ubuntu/clc/modules/storage-controller/native/Makefile
===================================================================
--- ubuntu.orig/clc/modules/storage-controller/native/Makefile	2009-09-03 18:47:33.671780949 -0500
+++ ubuntu/clc/modules/storage-controller/native/Makefile	2009-09-03 18:49:52.728777511 -0500
@@ -19,5 +19,5 @@
 	rm -f $(LVM2_SO) *.o *~* *#*  
 
 install:
-	mkdir -p $(EUCALYPTUS)/usr/lib/eucalyptus/
-	cp $(LIBDIR)/$(LVM2_SO) $(EUCALYPTUS)/usr/lib/eucalyptus/
+	mkdir -p $(DESTDIR)$(EUCALYPTUS)/usr/lib/eucalyptus/
+	cp $(LIBDIR)/$(LVM2_SO) $(DESTDIR)$(EUCALYPTUS)/usr/lib/eucalyptus/
Index: ubuntu/cluster/Makefile
===================================================================
--- ubuntu.orig/cluster/Makefile	2009-09-02 16:42:09.171016871 -0500
+++ ubuntu/cluster/Makefile	2009-09-03 18:49:52.748778309 -0500
@@ -59,20 +59,20 @@
 	$(CC) -o $(CLIENTKILLALL) $(CFLAGS) $(INCLUDES) $(CC_LIBS) generated/adb_*.o generated/axis2_stub_*.o  ../util/misc.o ../util/euca_auth.o $(CLIENT).c cc-client-marshal-adb.c -DMODE=0
 
 deploy:
-	$(INSTALL) -d $(AXIS2C_SERVICES)/$(SERVICE_NAME)/
-	$(INSTALL) $(SERVICE_SO) $(AXIS2C_SERVICES)/$(SERVICE_NAME)/
-	$(INSTALL) $(CCWSDL) $(AXIS2C_SERVICES)/$(SERVICE_NAME)/
+	$(INSTALL) -d $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/
+	$(INSTALL) $(SERVICE_SO) $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/
+	$(INSTALL) $(CCWSDL) $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/
 #no ws-sec
-	cat generated/services.xml | ../tools/add_wsdllocation.pl $(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(CCWSDLNAME) > $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml
+	cat generated/services.xml | ../tools/add_wsdllocation.pl $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(CCWSDLNAME) > $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml
 
 #ws-sec
-	cat generated/services.xml | ../tools/add_wsdllocation.pl $(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(CCWSDLNAME) | ../tools/add_service_policy.pl ../tools/service-policy-template.xml | sed "s:CLIENT-CERT:cloud-cert.pem:g" | sed "s:CLIENT-KEY:cloud-pk.pem:g" | sed "s:SERVER-CERT:cluster-cert.pem:g" | sed "s:SERVER-KEY:cluster-pk.pem:g" | sed "s:CLIENT-USERNAME:eucalyptus:g" > $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml
+	cat generated/services.xml | ../tools/add_wsdllocation.pl $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(CCWSDLNAME) | ../tools/add_service_policy.pl ../tools/service-policy-template.xml | sed "s:CLIENT-CERT:cloud-cert.pem:g" | sed "s:CLIENT-KEY:cloud-pk.pem:g" | sed "s:SERVER-CERT:cluster-cert.pem:g" | sed "s:SERVER-KEY:cluster-pk.pem:g" | sed "s:CLIENT-USERNAME:eucalyptus:g" > $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml
 	cat ../tools/client-policy-template.xml | sed "s:EUCALYPTUS_HOME:$(EUCALYPTUS):g" | sed "s:AXIS2C_HOME:$(AXIS2C_HOME):g" | sed "s:CLIENT-CERT:cloud-cert.pem:g" | sed "s:SERVER-CERT:cluster-cert.pem:g" | sed "s:SERVER-KEY:cluster-pk.pem:g" | sed "s:CLIENT-KEY:cloud-pk.pem:g" | sed "s:CLIENT-USERNAME:eucalyptus:g" > cc-client-policy.xml
 ##	cp ../tools/*.pem $(EUCALYPTUS)/var/lib/eucalyptus/keys/
 #	cp cc-client-policy.xml $(EUCALYPTUS)/var/lib/eucalyptus/keys/
 #	chmod -R 0600 $(EUCALYPTUS)/var/lib/eucalyptus/keys/*
 # ws
-	$(INSTALL) $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
+	$(INSTALL) $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
 # no ws
 #	cp $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
 
@@ -85,5 +85,5 @@
 	rm -f generated/stubs
 
 install: deploy
-	@$(INSTALL) -m 0555 $(CLIENTKILLALL) $(sbindir)
-	@$(INSTALL) cc-client-policy.xml $(vardir)/lib/eucalyptus/keys/
+	@$(INSTALL) -m 0555 $(CLIENTKILLALL) $(DESTDIR)$(sbindir)
+	@$(INSTALL) cc-client-policy.xml $(DESTDIR)$(vardir)/lib/eucalyptus/keys/
Index: ubuntu/gatherlog/Makefile
===================================================================
--- ubuntu.orig/gatherlog/Makefile	2009-09-02 16:42:09.195022227 -0500
+++ ubuntu/gatherlog/Makefile	2009-09-03 18:49:52.768795099 -0500
@@ -46,11 +46,11 @@
 	$(CC) -o $(CLIENT) $(CFLAGS) $(INCLUDES) $(LIBS) generated/adb_*.o generated/axis2_stub_*.o  ../util/misc.o ../util/euca_auth.o $(CLIENT).c gl-client-marshal-adb.c -DMODE=1
 
 deploy:
-	$(INSTALL) -d $(AXIS2C_SERVICES)/$(SERVICE_NAME)/
-	$(INSTALL) $(SERVICE_SO) $(AXIS2C_SERVICES)/$(SERVICE_NAME)/
-	$(INSTALL) $(GLWSDL) $(AXIS2C_SERVICES)/$(SERVICE_NAME)/
+	$(INSTALL) -d $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/
+	$(INSTALL) $(SERVICE_SO) $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/
+	$(INSTALL) $(GLWSDL) $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/
 #no ws-sec
-	cat generated/services.xml | ../tools/add_wsdllocation.pl $(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(GLWSDLNAME) > $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml
+	cat generated/services.xml | ../tools/add_wsdllocation.pl $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(GLWSDLNAME) > $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml
 
 #ws-sec
 #	cat generated/services.xml | ../tools/add_wsdllocation.pl $(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(GLWSDLNAME) | ../tools/add_service_policy.pl ../tools/service-policy-template.xml | sed "s:CLIENT-CERT:cloud-cert.pem:g" | sed "s:CLIENT-KEY:cloud-pk.pem:g" | sed "s:SERVER-CERT:cluster-cert.pem:g" | sed "s:SERVER-KEY:cluster-pk.pem:g" | sed "s:CLIENT-USERNAME:eucalyptus:g" > $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml
@@ -61,7 +61,7 @@
 # ws
 #	cp $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
 # no ws
-	$(INSTALL) $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
+	$(INSTALL) $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
 
 clean:
 	rm -f $(SERVICE_SO) *.o $(CLIENT) *~* *#*  
Index: ubuntu/node/Makefile
===================================================================
--- ubuntu.orig/node/Makefile	2009-09-02 16:42:09.215002212 -0500
+++ ubuntu/node/Makefile	2009-09-03 18:49:52.788800784 -0500
@@ -79,19 +79,19 @@
 	$(CC) $(CFLAGS) $(INCLUDES) $(NC_LIBS) -o test_nc test_nc.c ../util/misc.o
 
 deploy:
-	$(INSTALL) -d $(AXIS2C_SERVICES)/$(SERVICE_NAME)/
-	$(INSTALL) $(SERVICE_SO) $(AXIS2C_SERVICES)/$(SERVICE_NAME)/
+	$(INSTALL) -d $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/
+	$(INSTALL) $(SERVICE_SO) $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/
 
 # no ws-sec
-	cat generated/services.xml | ../tools/add_wsdllocation.pl $(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(NCWSDLNAME) > $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml
+	cat generated/services.xml | ../tools/add_wsdllocation.pl $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(NCWSDLNAME) > $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml
 # ws-sec
-	cat generated/services.xml | ../tools/add_wsdllocation.pl $(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(NCWSDLNAME) | ../tools/add_service_policy.pl ../tools/service-policy-template.xml | sed "s:CLIENT-CERT:cluster-cert.pem:g" | sed "s:CLIENT-KEY:cluster-pk.pem:g" | sed "s:SERVER-CERT:node-cert.pem:g" | sed "s:SERVER-KEY:node-pk.pem:g" | sed "s:CLIENT-USERNAME:eucalyptus:g" > $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml
+	cat generated/services.xml | ../tools/add_wsdllocation.pl $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/$(NCWSDLNAME) | ../tools/add_service_policy.pl ../tools/service-policy-template.xml | sed "s:CLIENT-CERT:cluster-cert.pem:g" | sed "s:CLIENT-KEY:cluster-pk.pem:g" | sed "s:SERVER-CERT:node-cert.pem:g" | sed "s:SERVER-KEY:node-pk.pem:g" | sed "s:CLIENT-USERNAME:eucalyptus:g" > $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml
 	cat ../tools/client-policy-template.xml | sed "s:EUCALYPTUS_HOME:$(EUCALYPTUS):g" | sed "s:AXIS2C_HOME:$(AXIS2C_HOME):g"  | sed "s:CLIENT-CERT:cluster-cert.pem:g" | sed "s:CLIENT-KEY:cluster-pk.pem:g" | sed "s:SERVER-CERT:node-cert.pem:g" | sed "s:SERVER-KEY:node-pk.pem:g" | sed "s:CLIENT-USERNAME:eucalyptus:g" > nc-client-policy.xml
 #	cp nc-client-policy.xml $(EUCALYPTUS)/var/lib/eucalyptus/keys/
 ##	cp ../tools/*.pem $(EUCALYPTUS)/var/lib/eucalyptus/keys/
 #	chmod -R 0600 $(EUCALYPTUS)/var/lib/eucalyptus/keys/*
 # ws
-	$(INSTALL) $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
+	$(INSTALL) $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_WS.xml $(DESTDIR)$(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
 # no ws
 #	cp $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services_noWS.xml $(AXIS2C_SERVICES)/$(SERVICE_NAME)/services.xml
 
@@ -105,5 +105,5 @@
 	rm -rf $(SERVICE_SO) *.o $(CLIENT) $(CLIENT)_local *~* *#*
 
 install: deploy
-	$(INSTALL) nc-client-policy.xml $(vardir)/lib/eucalyptus/keys/
-	$(INSTALL) -m 0755 test_nc $(sbindir)/euca_test_nc
+	$(INSTALL) nc-client-policy.xml $(DESTDIR)$(vardir)/lib/eucalyptus/keys/
+	$(INSTALL) -m 0755 test_nc $(DESTDIR)$(sbindir)/euca_test_nc
Index: ubuntu/tools/Makefile
===================================================================
--- ubuntu.orig/tools/Makefile	2009-09-03 18:47:34.515781456 -0500
+++ ubuntu/tools/Makefile	2009-09-03 18:49:52.788800784 -0500
@@ -34,30 +34,30 @@
 	@rm -rf eucalyptus eucalyptus-cloud eucalyptus-walrus eucalyptus-sc
 
 install: build
-	@if test -e /etc/init.d/eucalyptus ; then \
-		mv /etc/init.d/eucalyptus /etc/init.d/eucalyptus.old ; \
-		echo "Moving away old init.d script"; fi
-	@if test -e $(etcdir)/init.d/eucalyptus ; then \
-		mv $(etcdir)/init.d/eucalyptus $(etcdir)/init.d/eucalyptus.old ; fi
-	@$(INSTALL) -m 755 eucalyptus-cloud $(etcdir)/init.d
-	@$(INSTALL) -m 755 eucalyptus-walrus $(etcdir)/init.d
-	@$(INSTALL) -m 755 eucalyptus-sc $(etcdir)/init.d
-	@$(INSTALL) -m 755 eucalyptus-nc $(etcdir)/init.d
-	@$(INSTALL) -m 755 eucalyptus-cc $(etcdir)/init.d
-	@$(INSTALL) -m 644 httpd.conf $(etcdir)/eucalyptus
-	@if test -e $(etcdir)/eucalyptus/eucalyptus.conf ; then \
-		cp $(etcdir)/eucalyptus/eucalyptus.conf $(etcdir)/eucalyptus/eucalyptus.conf.old ; fi
-	@$(INSTALL) -m 644 eucalyptus.conf $(etcdir)/eucalyptus
-	@$(INSTALL) -m 644 vtunall.conf.template $(etcdir)/eucalyptus
-	@$(INSTALL) -m 755 add_key.pl $(datarootdir)/eucalyptus
-	@$(INSTALL) -m 755 gen_libvirt_xml $(datarootdir)/eucalyptus
-	@$(INSTALL) -m 755 gen_kvm_libvirt_xml $(datarootdir)/eucalyptus
-	@$(INSTALL) -m 755 detach.pl $(datarootdir)/eucalyptus
-	@$(INSTALL) -m 755 partition2disk $(datarootdir)/eucalyptus
-	@$(INSTALL) -m 755 get_xen_info $(datarootdir)/eucalyptus/get_xen_info
-	@$(INSTALL) -m 755 get_sys_info $(datarootdir)/eucalyptus/get_sys_info
-	@$(INSTALL) -m 755 euca_ipt $(datarootdir)/eucalyptus
-	@$(INSTALL) -m 755 populate_arp.pl $(datarootdir)/eucalyptus
-	@$(INSTALL) -m 755 euca_conf $(sbindir)
-	@$(INSTALL) -m 755 euca_sync_key $(sbindir)
+#	@if test -e /etc/init.d/eucalyptus ; then \
+#		mv /etc/init.d/eucalyptus /etc/init.d/eucalyptus.old ; \
+#		echo "Moving away old init.d script"; fi
+#	@if test -e $(etcdir)/init.d/eucalyptus ; then \
+#		mv $(etcdir)/init.d/eucalyptus $(etcdir)/init.d/eucalyptus.old ; fi
+	@$(INSTALL) -m 755 eucalyptus-cloud $(DESTDIR)$(etcdir)/init.d
+	@$(INSTALL) -m 755 eucalyptus-walrus $(DESTDIR)$(etcdir)/init.d
+	@$(INSTALL) -m 755 eucalyptus-sc $(DESTDIR)$(etcdir)/init.d
+	@$(INSTALL) -m 755 eucalyptus-nc $(DESTDIR)$(etcdir)/init.d
+	@$(INSTALL) -m 755 eucalyptus-cc $(DESTDIR)$(etcdir)/init.d
+	@$(INSTALL) -m 644 httpd.conf $(DESTDIR)$(etcdir)/eucalyptus
+	@if test -e $(DESTDIR)$(etcdir)/eucalyptus/eucalyptus.conf ; then \
+		cp $(etcdir)/eucalyptus/eucalyptus.conf $(DESTDIR)$(etcdir)/eucalyptus/eucalyptus.conf.old ; fi
+	@$(INSTALL) -m 644 eucalyptus.conf $(DESTDIR)$(etcdir)/eucalyptus
+	@$(INSTALL) -m 644 vtunall.conf.template $(DESTDIR)$(etcdir)/eucalyptus
+	@$(INSTALL) -m 755 add_key.pl $(DESTDIR)$(datarootdir)/eucalyptus
+	@$(INSTALL) -m 755 gen_libvirt_xml $(DESTDIR)$(datarootdir)/eucalyptus
+	@$(INSTALL) -m 755 gen_kvm_libvirt_xml $(DESTDIR)$(datarootdir)/eucalyptus
+	@$(INSTALL) -m 755 detach.pl $(DESTDIR)$(datarootdir)/eucalyptus
+	@$(INSTALL) -m 755 partition2disk $(DESTDIR)$(datarootdir)/eucalyptus
+	@$(INSTALL) -m 755 get_xen_info $(DESTDIR)$(datarootdir)/eucalyptus/get_xen_info
+	@$(INSTALL) -m 755 get_sys_info $(DESTDIR)$(datarootdir)/eucalyptus/get_sys_info
+	@$(INSTALL) -m 755 euca_ipt $(DESTDIR)$(datarootdir)/eucalyptus
+	@$(INSTALL) -m 755 populate_arp.pl $(DESTDIR)$(datarootdir)/eucalyptus
+	@$(INSTALL) -m 755 euca_conf $(DESTDIR)$(sbindir)
+	@$(INSTALL) -m 755 euca_sync_key $(DESTDIR)$(sbindir)
 
Index: ubuntu/clc/build.xml
===================================================================
--- ubuntu.orig/clc/build.xml	2009-09-02 16:42:09.151024246 -0500
+++ ubuntu/clc/build.xml	2009-09-03 18:49:52.867778284 -0500
@@ -163,27 +163,27 @@
   <property name="euca.log.dir" value="${euca.home}/var/log/eucalyptus"/>
   <target name="install">
     <!-- copy libraries over -->
-    <mkdir dir="${euca.lib.dir}"/>
-    <copy todir="${euca.lib.dir}">
+    <mkdir dir="${localenv.DESTDIR}${euca.lib.dir}"/>
+    <copy todir="${localenv.DESTDIR}${euca.lib.dir}">
       <fileset dir="${basedir}/target">
         <include name="**/*.jar"/>
       </fileset>
     </copy>
-    <copy todir="${euca.lib.dir}">
+    <copy todir="${localenv.DESTDIR}${euca.lib.dir}">
       <fileset dir="${deps.lib.dir}">
         <include name="**/*"/>
       </fileset>
     </copy>
-    <mkdir dir="${euca.conf.dir}"/>
-    <chmod dir="${euca.conf.dir}" perm="0755"/>
-    <mkdir dir="${euca.var.dir}/"/>
-    <mkdir dir="${euca.var.dir}/db"/>
-    <mkdir dir="${euca.var.dir}/keys"/>
+    <mkdir dir="${localenv.DESTDIR}${euca.conf.dir}"/>
+    <chmod dir="${localenv.DESTDIR}${euca.conf.dir}" perm="0755"/>
+    <mkdir dir="${localenv.DESTDIR}${euca.var.dir}/"/>
+    <mkdir dir="${localenv.DESTDIR}${euca.var.dir}/db"/>
+    <mkdir dir="${localenv.DESTDIR}${euca.var.dir}/keys"/>
     <!--	<chmod dir="${euca.var.dir}/db" perm="0700"/> 
 	<chmod dir="${euca.var.dir}/keys" perm="0700"/> -->
-    <mkdir dir="${euca.log.dir}"/>
-    <mkdir dir="${euca.var.dir}/webapps"/>
-    <mkdir dir="${euca.var.dir}/modules"/>
+    <mkdir dir="${localenv.DESTDIR}${euca.log.dir}"/>
+    <mkdir dir="${localenv.DESTDIR}${euca.var.dir}/webapps"/>
+    <mkdir dir="${localenv.DESTDIR}${euca.var.dir}/modules"/>
     <ant dir="modules/msgs" inheritall="false" target="install"/>
     <ant dir="modules/cloud" inheritall="false" target="install"/>
     <ant dir="modules/authentication" inheritall="false" target="install"/>
Index: ubuntu/clc/modules/www/build.xml
===================================================================
--- ubuntu.orig/clc/modules/www/build.xml	2009-09-02 16:42:09.131001310 -0500
+++ ubuntu/clc/modules/www/build.xml	2009-09-03 18:49:52.879779043 -0500
@@ -135,7 +135,7 @@
 
 	<property name="servlet.dir" value="${euca.var.dir}/webapps" />
 	<target name="install" depends="copy-conf">
-		<copy file="${gwt.war}" todir="${servlet.dir}" />
+		<copy file="${gwt.war}" todir="${localenv.DESTDIR}${servlet.dir}" />
 	</target>
 
 
